WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift

```diff
@@ -982,7 +985,35 @@ open class ReaderDetailViewController: UIViewController, UIViewControllerRestora
 
 
     @IBAction func didTapMenuButton(_ sender: UIButton) {
-        ReaderPostMenu.showMenuForPost(post!, fromView: menuButton, inViewController: self)
+        guard let post = post,
+            let context = post.managedObjectContext else {
+            return
+        }
+
+        guard post.isFollowing else {
+            ReaderPostMenu.showMenuForPost(post, fromView: menuButton, inViewController: self)
+            return
+        }
+
+        let service = ReaderTopicService(managedObjectContext: context)
+        if let topic = service.findSiteTopic(withSiteID: post.siteID) {
+            ReaderPostMenu.showMenuForPost(post, topic: topic, fromView: menuButton, inViewController: self)
+            return
+        }
+
+        SVProgressHUD.show()
```

Comment:
Could you perhaps extract this last part of the method into a separate method, named to clarify what it's doing? I think it's... if there's no site topic, we fetch and try again?

Reply:
Good catch because actually this is not needed anymore